### ISSUE TITLE
YOLOv5l faster than YOLOv5m

### ISSUE THREAD

**Issue by JonathanSamelson** [ultralytics/yolov5#7590]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi,

After training YOLOv5 models on the dataset [MIO-TCD](https://tcd.miovision.com/), I did several inference tests to see the speed of my models. Surprisingly, YOLOv5l is faster than YOLOv5m when doing inference on the validation set (11.000 images). 
Here's the results of my test:

| Model | Speed (FPS) |
|:--------------:|:------:|
|    Nano | 61,11 |
|    Small | 61,72 |
|    Medium | **33,98** |
|    Large | **38,49** |
|    XL  | 27,75 |

I use the following environment: `YOLOv5  2022-4-14 torch 1.10.2+cu113 CUDA:0 (GeForce GTX 1080 Ti, 11264.0MB)`

The same phenomenon can be seen with nano and small but this is only a slight difference that does not seem significative. In particular, I found [this post](https://community.ultralytics.com/t/yolov5-study-batch-size-vs-speed/31) written in February where YOLOv5n and YOLOv5s almost share the same inference speed.

However I ran the medium and large models and I can always see a ~5 FPS difference, which is far more significative if we compare to the above post.

Here is the code I used for my test:
```
model_weights = "yolov5m-mio.pt"
# model_weights = "yolov5l-mio.pt"
net = torch.hub.load('ultralytics/yolov5:v6.0', 'custom', path=model_weights, verbose=False)
net.cuda()
net.conf = 0.2
net.iou = 0.45
net.agnostic = False

for img_path in tqdm(list_test_img_path):
    frame = cv2.imread(img_path)
    output = net(frame[..., ::-1], size=416)  # BGR to RGB
    results = np.array(output.xyxy[0].cpu())
```

Do you any idea why YOLOv5l can be faster than YOLOv5m?
Thank you in advance for your response.

**Comment by abuelgasimsaadeldin** [comment]
Hi @JonathanSamelson,

I've recently also downloaded the "MIO-TCD Localization" dataset and I've found that the ground truth annotations were given in a CSV file, may I know how you converted that into TXT annotation files for YOLOv5 training? I would like to also run similar tests to confirm your results. Thanks and best regards!

Regards,
Abu

**Comment by glenn-jocher** [comment]
@JonathanSamelson official profiling results are here for all models: https://github.com/ultralytics/yolov5#pretrained-checkpoints

Your results may vary by hardware, software, drivers, etc.

**Comment by JonathanSamelson** [comment]
Hi @abuelgasimsaadeldin,

It has been a while I converted the CSV files for training. Fortunately I think I found the code I used for conversion:
[format_mio-tcd_yolo.zip](https://github.com/ultralytics/yolov5/files/8571934/format_mio-tcd_yolo.zip)

Here you can also find the TXT annotations directly (hopefully you should get the same results using my script):
[labels-MIO-TCD-Localization.zip](https://github.com/ultralytics/yolov5/files/8571856/labels.zip)

Don't hesitate to share with me your results! 😉

Hi @glenn-jocher, yeah that's what I thought. It's probably related to those. I noticed that using another machine, my YOLOv5l is a bit slower than my YOLOv5m. Still, such a difference makes me a bit confused.

If anyone that has a 1080Ti at his disposal is willing to share some benchmarks it could probably help identifying the issue.

**Comment by abuelgasimsaadeldin** [comment]
Hi @JonathanSamelson,

Thanks a lot for that I really appreciate it. I have an RTX 3060 laptop GPU and will definitely be running some experiments on this dataset sometime soon and will update you on the results hopefully, thanks again!

Regards,
Abu

**Comment by yin-qiyu** [comment]
Hi @JonathanSamelson,
How is your FPS calculated?

**Comment by JonathanSamelson** [comment]
Hi @yin-qiyu,
The FPS are calculated by the tqdm module (see the code snippet above) which is used to iterate over the validation dataset.

Therefore, it takes into account the reading time, but this should be the same for each model.
